oras/auth/__init__.py

```diff
@@ -14,10 +14,11 @@ class AuthenticationException(Exception):
     pass
 
 
-def get_auth_backend(name="token", session=None, **kwargs):
+def get_auth_backend(name="token", session=None, prefix="https", **kwargs):
```

Comment:
@tarilabs this won't always be https - specifically in testing. I think we'd want to inherit this from the parent that created it, and use the same convention there with insecure as a boolean to determine the prefix (and not hard code http or https). Let me know your thoughts.

Reply:
this line is specifically about the default isn't it,
and is mainly never used because here:

https://github.com/oras-project/oras-py/pull/165/files#diff-ccb407ceff93417721d33c41bf1975c04c98454fd2a36518a7b8572808db03a4R81

is set explicitly. 

am I missing something? 🤔

Reply:
Yes I think you want to set it based on the insecure parameter:

```
 self.prefix: str = "http" if insecure else "https"
```
Which is in the init of the provider.

Reply:
mmm, I've actually mimic-ed on existing code and did that in line 23 of this file.

making the line you quote:

https://github.com/oras-project/oras-py/pull/165/files#diff-fcbd6629a667f0deb29e01bccd3fa21f8790961f48092d8cee24ad00986706a9R22

another (unused) default.

I'm happy to refactor this code differently if that helps, I was just trying to go with the existing "code convention"? :)